### PR TITLE
Add Cmdline Redaction support

### PR DESCRIPTION
**Pull request by taglines**

### CODE REVIEW

File: auoms.cpp [hunk 1]
```diff
@@ -148,6 +149,10 @@ int main(int argc, char**argv) {
         rules_dir = config.GetString("rules_dir");
     }
 
+    if (config.HasKey("redact_dir")) {
+        rules_dir = config.GetString("redact_dir");
```
Comment (bmanikm) [inline auoms.cpp:153]: >rules_dir [](start = 8, length = 9)

this needs to be redact_dir, right?

File: CmdlineRedactor.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+#include "StringUtils.h"
+
+std::shared_ptr<CmdlineRedactionRule> CmdlineRedactionRule::LoadFromFile(const std::string& path) {
+    Config config;
+    try {
+        config.Load(path);
+
+        std::string replacement_char = "*";
+        if (config.HasKey("replacement_char")) {
+            replacement_char = config.GetString("replacement_char");
```
Comment (bmanikm) [inline CmdlineRedactor.cpp:31]: >replacement_char [](start = 12, length = 16)

Do we need to warn if more than one char specified as value for replacement_char?

File: CmdlineRedactor.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        if (ends_with(name, ".conf")) {
+            auto rule = CmdlineRedactionRule::LoadFromFile(dir + "/" + name);
+            if (rule) {
+                _rules.emplace_back(rule);
+            }
+        }
+    }
+}
+
+bool CmdlineRedactor::Check(std::string& cmdline) const {
```
Comment (bmanikm) [inline CmdlineRedactor.cpp:105]: >Check [](start = 22, length = 5)

As check() is updating/filling the given cmdline string after checking, consider renaming the function, may be rewrite, redact, or other suitable name?

File: auoms.cpp (hunk 1 above)
Comment (taglines) [reply auoms.cpp]: Good catch. Fixed

File: CmdlineRedactor.cpp (hunk 3 above)
Comment (taglines) [reply CmdlineRedactor.cpp]: Renamed.

**Review by bmanikm — APPROVED** [review]
:shipit:

**Review by sethRait — APPROVED** [review]

**Pull request merged**